benchmarks/sparse/plot_results.py

```diff
@@ -0,0 +1,61 @@
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+from pathlib import Path
+import pickle
+import argparse
+
```

Comment:
Add a comment here explaining what this file is for and how it's intended to be used.

In the interest of avoiding this PR being too massive, I suggest separating the benchmark files into a separate PR.